app/route53.py

```diff
@@ -0,0 +1,248 @@
+    record : dict
+        The ``ResourceRecordSet`` with a ``Name`` corresponding to the
+        `cname_domain`. E.g.,
+
+        .. code-block:: python
+
+           [{'Name': 'www.mydomain.org.',
+             'ResourceRecords': [{'Value': 'origin_domain.com'}],
+             'TTL': 14400,
+             'Type': 'CNAME'}]
+
+        `None` is return is not ResourceRecordSet matching `cname_domain` is
+        found.
+    """
+    # turns out boto3 doesn't need StardRecordName in lexicographic order
+    # despite their docs.
+    # name = _lexicographic_order_domain(cname_url)
+
+    if not cname_domain.endswith('.'):
+        cname_domain = cname_domain + '.'
+
+    r = client.list_resource_record_sets(
+        HostedZoneId=zone_id,
+        StartRecordName=cname_domain,
+        StartRecordType='CNAME'
+    )
+    if r['ResponseMetadata']['HTTPStatusCode'] != 200:
+        msg = 'list_resource_record_sets failed:\n' + pformat(r)
+        log.error(msg)
+        raise Route53Error(msg)
+    for record in r['ResourceRecordSets']:
+        if record['Name'] == cname_domain:
+            log.info(pformat(record))
+            return record
+
+    log.info('No existing CNAME record found for {0}'.format(cname_domain))
+    return None
+
+
+def _upsert_cname_record(client, zone_id, cname_domain, origin_domain):
+    """Upsert a CNAME record of `cname_domain` that points to `origin_domain`.
+    """
+    change = {
+        'Action': 'UPSERT',
+        'ResourceRecordSet': {
+            'Name': cname_domain,
+            'Type': 'CNAME',
+            'TTL': 900,
+            'ResourceRecords': [
+                {
+                    'Value': origin_domain
+                },
+            ],
+        }
+    }
+    change_batch = {
+        'Comment': 'Upsert {0} -> {1}'.format(cname_domain, origin_domain),
+        'Changes': [change]}
+    log.info(pformat(change_batch))
+
+    r = client.change_resource_record_sets(
+        HostedZoneId=zone_id,
+        ChangeBatch=change_batch)
+    log.info(r)
+    if r['ResponseMetadata']['HTTPStatusCode'] != 200:
+        msg = 'change_resource_record_sets failed with:\n' + pformat(change)
+        log.error(msg)
+        raise Route53Error(msg)
+
+
+def _derive_fully_specified_domain(url):
+    """From a URL, get the fully-specified domain. These are used as names
+    for hosted zones in Route 53.
+
+    For example, the fully-specified domain of `https://www.domain.org` is
+    `domain.org.` (note the trailing period).
+    """
+    if '//' not in url:
+        url = '//' + url
+    p = urlparse(url)
+    domain = p.netloc
+    domain = domain.rstrip('.')  # in case we're working with an FSD originally
+    if p.port is not None:
+        domain, _ = domain.split(':')
+    fsd = '.'.join(domain.split('.')[-2:]) + '.'
+    return fsd
+
+
+def _lexicographic_order_domain(url):
+    """Order a url by lexicographic order.
+
+    E.g. `community.lsst.org` becomes `org.lsst.community`.
+    """
```

Comment:
Could you call ` _derive_fully_specified_domain` instead of implementing the same logic?